### ISSUE TITLE
Jupyter notebook Kernel failed to start after env update...

### ISSUE THREAD

**Issue by AirVetra** [fastai/fastai#846]
On both PCs with fastai and fastai-cpu env, with clean install kernel failed to start... I make full clean reinstall of anaconda and then fastai as described in "old" version... Spend half of the day, have no idea adn no clue how to solve the problem (google search hasn't helped)...

Windows 10 x64 (1803 and 1809) , Anaconda 5.3 (latest)

Traceback (most recent call last):
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\web.py", line 1512, in _execute
    result = yield result
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\notebook\services\sessions\handlers.py", line 73, in post
    type=mtype))
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\notebook\services\sessions\sessionmanager.py", line 79, in create_session
    kernel_id = yield self.start_kernel_for_session(session_id, path, name, type, kernel_name)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\notebook\services\sessions\sessionmanager.py", line 92, in start_kernel_for_session
    self.kernel_manager.start_kernel(path=kernel_path, kernel_name=kernel_name)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\tornado\gen.py", line 307, in wrapper
    yielded = next(result)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\notebook\services\kernels\kernelmanager.py", line 160, in start_kernel
    super(MappingKernelManager, self).start_kernel(**kwargs)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\jupyter_client\multikernelmanager.py", line 110, in start_kernel
    km.start_kernel(**kwargs)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\jupyter_client\manager.py", line 259, in start_kernel
    **kw)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\jupyter_client\manager.py", line 204, in _launch_kernel
    return launch_kernel(kernel_cmd, **kw)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\site-packages\jupyter_client\launcher.py", line 128, in launch_kernel
    proc = Popen(cmd, **kwargs)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "C:\Users\airve\Anaconda3\envs\fastai-cpu\lib\subprocess.py", line 997, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] Не удается найти указанный файл

**Comment by AirVetra** [comment]
I made it!!!

in C:\Users\airve\Anaconda3\envs\fastai-cpu\share\jupyter\kernels\python3\kernel.json

there was a misleading path to python!!! Made it bold - just deleted it and everything works!!!
So, seems someone changed it in the script and not right path is written... 

Pls. check my solution...

{
 "argv": [
  "C:/Users/airve/Anaconda3/envs/fastai-cpu/**bin/**python",
  "-m",
  "ipykernel_launcher",
  "-f",
  "{connection_file}"
 ],
 "display_name": "Python 3",
 "language": "python"
}

**Comment by stas00** [comment]
Thank you for giving the exact context for the problem, @AirVetra. I now understand your report.

So as you discovered the problem is that your anaconda + jupyter set incorrectly the path to python in `jupyter\kernels\python3\kernel.json`. I'm pretty sure that if you repeat the same setup and don't involve fastai at all (i.e. just install anaconda + jupyter) you will have exactly the same issue. fastai is just a pure python module that can run on top of jupyter notebook but otherwise it doesn't touch the jupyter setup at all.

So you probably need to take this to jupyter team, and a quick search shows that other people have similar issues, e.g. [this](https://github.com/jupyter/notebook/issues/2563) - there are a few proposed solutions in there.

**Comment by danieljue** [comment]
Sharing this for the benefit of FastAI users on Windows.  This happened to me as well tonight, Windows 10 fresh install, most up-to-date Anaconda3 as of 11/20/18.  Fastai is the only env I've installed so far, but like stas00 said, it will probably affect others as well.  I have **not** had this issue on Mac nor on EC2 Ubuntu. In the kernel.json there was an extra "/bin" in the path before python, which caused jupyter to be unable to start a notebook kernel.

I changed my kernel.json to look like this and it works, i.e. remove the offending /bin before the python.
`{
 "argv": [
  "C:/Users/MyUserName/AppData/Local/conda/conda/envs/fastai/python",
  "-m",
  "ipykernel_launcher",
  "-f",
  "{connection_file}"
 ],
 "display_name": "Python 3",
 "language": "python"
}`
